src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -565,14 +568,38 @@ def test_fw_binaryoperator(fort_writer):
         "subroutine tmp(a,n)\n"
         "  integer, intent(in) :: n\n"
         "  real, intent(out) :: a(n)\n"
-        "    a = sign(1.0,1.0)\n"
+        "    a = {0}(1.0,1.0)\n"
+        "end subroutine tmp\n"
+        "end module test").format(binary_intrinsic)
+    schedule = create_schedule(code, "tmp")
+
+    # Generate Fortran from the PSyIR schedule
+    result = fort_writer(schedule)
+    assert "a={0}(1.0, 1.0)".format(binary_intrinsic.upper()) in result
```

Comment:
We could add a compilation test here but I'm not sure it's strictly necessary (as the code should be unchanged from the input).

Reply:
I agree. It makes sense to test the fortran code produced by the fortran backend. I'm a bit embarrassed that this is not already done. Anyway added it for this and other operators tests.